app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/SettingsValues.java

```diff
@@ -149,7 +149,7 @@ public SettingsValues(final Context context, final SharedPreferences prefs, fina
         mKeyPreviewPopupOn = Settings.readKeyPreviewPopupEnabled(prefs, res);
         mSlidingKeyInputPreviewEnabled = prefs.getBoolean(
                 DebugSettings.PREF_SLIDING_KEY_INPUT_PREVIEW, true);
-        mShowsVoiceInputKey = needsToShowVoiceInputKey(prefs, res) && mInputAttributes.mShouldShowVoiceInputKey;
+        mShowsVoiceInputKey = needsToShowVoiceInputKey(prefs, res) && mInputAttributes.mShouldShowVoiceInputKey && RichInputMethodManager.getInstance().hasShortcutIme();
```

Comment:
Maybe the check should be included earlier into `mInputAttributes.mShouldShowVoiceInputKey`, i.e. in [`InputAttributes`](https://github.com/openboard-team/openboard/blob/7fdb5954d018372f47a18e1e25093405aa0d1e21/app/src/main/java/org/dslul/openboard/inputmethod/latin/InputAttributes.java#L112) ?

Reply:
You're right, this is a more "natural" position